lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -202,7 +209,7 @@ SerializedModuleLoaderBase::findModule(AccessPathElem moduleID,
     // We can only get here if all targetFileNamePairs failed with
     // 'std::errc::no_such_file_or_directory'.
     if (maybeDiagnoseArchitectureMismatch(moduleID.second,
```

Comment:
The error message from `maybeDiagnoseArchitectureMismatch()` should be changed to say "for target" instead of "for architecture". Arguably the method's name should change too.